users/rs/rs.c

```diff
@@ -0,0 +1,90 @@
+
+#include "rs.h"
+
+__attribute__ ((weak))
+bool process_record_keymap(uint16_t keycode, keyrecord_t *record) {
+  return true;
+}
+
+__attribute__ ((weak))
+void rgb_mod_changed_keymap(void) {
+}
+
+__attribute__ ((weak))
+void keylog_set_keymap(uint16_t keycode, keyrecord_t *record) {
+}
+
+void press_key(uint16_t key) {
+  register_code(key);
+  unregister_code(key);
+}
+
+void press_two_keys(uint16_t key1, uint16_t key2) {
+  register_code(key1);
+  register_code(key2);
+  unregister_code(key2);
+  unregister_code(key1);
+}
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  if (record->event.pressed) {
+    keylog_set_keymap(keycode, record);
+  }
+
+  switch (keycode) {
+  case NEQL: // !=
+    if (record->event.pressed) {
+      SEND_STRING("!=");
+    }
+    return false;
+  case LTGT: // > or < with shift
+    if (record->event.pressed) {
+      if (get_mods() & (MOD_BIT(KC_LSFT) | MOD_BIT(KC_RSFT))) {
+        // <
+        press_key(KC_COMM); // shift is already registered
+      } else {
+        // >
+        press_two_keys(KC_LSFT, KC_DOT);
+      }
+    }
+    return false;
+  case LPLT: // ( or < with shift
+    if (record->event.pressed) {
+      if (get_mods() & (MOD_BIT(KC_LSFT) | MOD_BIT(KC_RSFT))) {
+        // <
+        press_key(KC_COMM); // shift is already registered
+      } else {
+        // (
+        press_two_keys(KC_LSFT, KC_9);
+      }
+    }
+    return false;
+  case RPGT: // ) or > with shift
+    if (record->event.pressed) {
+      if (get_mods() & (MOD_BIT(KC_LSFT) | MOD_BIT(KC_RSFT))) {
+        // <
+        press_key(KC_DOT); // shift is already registered
+      } else {
+        // )
+        press_two_keys(KC_LSFT, KC_0);
+      }
+    }
+    return false;
+#ifdef RGBLIGHT_ENABLE
+  case RGB_MOD:
+      if (record->event.pressed) {
+        rgblight_step();
```

Comment:
Actually, if you return true, and check on `!record->event.pressed`, you shouldn't need this, I think

Reply:
The problem is that `rgb_mod_changed_keymap` has to be call after the fact.

Reply:
Well, you could use `!record->event.pressed` instead, so it happens on upstroke, but I can understand that